src/analysisd/decoders/syscheck.c

```diff
@@ -1685,44 +1679,45 @@ int fim_fetch_attributes_state(cJSON *attr, Eventinfo *lf, char new_state) {
                 if (new_state) {
                     lf->fields[FIM_SIZE].value = w_long_str((long) attr_it->valuedouble);
                 } else {
-                    lf->size_before = w_long_str((long) attr_it->valuedouble);
+                    lf->fields[FIM_SIZE_BEFORE].value = w_long_str((long) attr_it->valuedouble);
                 }
             } else if (!strcmp(attr_it->string, "inode")) {
                 if (new_state) {
                     lf->fields[FIM_INODE].value = w_long_str((long) attr_it->valuedouble);
-                    lf->inode_after = (long) attr_it->valuedouble;
                 } else {
-                    lf->inode_before = (long) attr_it->valuedouble;
+                    lf->fields[FIM_INODE_BEFORE].value = w_long_str((long) attr_it->valuedouble);;
                 }
             } else if (!strcmp(attr_it->string, "mtime")) {
+                aux_time = (long) attr_it->valuedouble;
+                time_string = ctime_r(&aux_time, buf_ptr);
+                time_string[strlen(time_string) - 1] = '\0';
                 if (new_state) {
                     lf->fields[FIM_MTIME].value = w_long_str((long) attr_it->valuedouble);
-                    lf->mtime_after = (long)attr_it->valuedouble;
                 } else {
-                    lf->mtime_before = (long) attr_it->valuedouble;
+                    lf->fields[FIM_MTIME_BEFORE].value = w_long_str((long) attr_it->valuedouble);
                 }
             }
         } else if (attr_it->type == cJSON_String) {
             char **dst_data = NULL;
 
             if (!strcmp(attr_it->string, "perm")) {
```

Comment:
Can you change these `strcmp` to match the same structure? 
```suggestion
            if (strcmp(attr_it->string, "perm") == 0) {
```